master/buildbot/config.py

```diff
@@ -297,8 +297,13 @@ def copy_str_param(name, alt_key=None):
 
         if 'logCompressionMethod' in config_dict:
             logCompressionMethod = config_dict.get('logCompressionMethod')
-            if logCompressionMethod not in ('bz2', 'gz'):
-                error("c['logCompressionMethod'] must be 'bz2' or 'gz'")
+            if logCompressionMethod not in ('raw', 'bz2', 'gz', 'lz4'):
+                error("c['logCompressionMethod'] must be 'raw', 'bz2', 'gz' or 'lz4'")
+            elif logCompressionMethod == "lz4":
+                try:
+                    import lz4
+                except ImportError:
+                    error("To set c['logCompressionMethod'] to 'lz4' you must install the lz4 library ('pip install lz4')")
             self.logCompressionMethod = logCompressionMethod
```

Comment:
And so replace error("...") to log.msg => don't you think that the user may not see the error and continue running the master thinking that the log is compressed?
Stopping is more radical but more "informative".